dumb-init.c

```diff
@@ -153,9 +129,22 @@ int main(int argc, char *argv[]) {
 
         execvp(argv[1], &argv[1]);
     } else {
+        pid_t killed_pid;
+        int exit_status, status;
+
         DEBUG("Child spawned with PID %d.\n", child_pid);
-        for (;;) {
-            pause();
+
+        while ((killed_pid = waitpid(-1, &status, 0))) {
```

Comment:
The opengroup doc seems to indicate that waitpid(-1) and a SIGCHLD handler won't play nice.
http://pubs.opengroup.org/onlinepubs/9699919799/functions/wait.html#tag_16_645_08

The glibc doc seems to agree:
http://www.gnu.org/software/libc/manual/html_node/Job-Control-Signals.html#index-SIGCHLD

Let's put this back (`for(;;)pause()`), and put the new SIGTERM code in the signal handler.